contracts/external/nexus/WrappedCoveredPrincipalToken.sol

```diff
@@ -0,0 +1,238 @@
+        _trancheFactory = __trancheFactory;
+        _trancheBytecodeHash = __trancheBytecodeHash;
+        _setupRole(ADMIN_ROLE, _owner);
+        _setRoleAdmin(ADMIN_ROLE, ADMIN_ROLE);
+        _setRoleAdmin(RECLAIM_ROLE, ADMIN_ROLE);
+    }
+
+    ///@notice Allows to create the name for the wrapped token.
+    function _processName(string memory _tokenSymbol)
+        internal
+        pure
+        returns (string memory)
+    {
+        return
+            string(
+                abi.encodePacked("Wrapped", _tokenSymbol, "Covered Principal")
+            );
+    }
+
+    ///@notice Allows to create the symbol for the wrapped token.
+    function _processSymbol(string memory _tokenSymbol)
+        internal
+        pure
+        returns (string memory)
+    {
+        return string(abi.encodePacked("W", _tokenSymbol));
+    }
+
+    /// @notice Add wrapped position within the allowed wrapped position enumerable set.
+    /// @dev    It is only allowed to execute by the owner of the contract.
+    ///         wrapped position which has underlying token equals to the base token are
+    ///         only allowed to add, Otherwise it will revert.
+    /// @param  _wrappedPosition Address of the Wrapped position which needs to add.
+    function addWrappedPosition(address _wrappedPosition)
+        external
+        override
+        isValidWp(_wrappedPosition)
+        onlyRole(ADMIN_ROLE)
+    {
+        require(
+            address(IWrappedPosition(_wrappedPosition).token()) == baseToken,
+            "WFP:INVALID_WP"
+        );
+        _allowedWrappedPositions.add(_wrappedPosition);
+        emit WrappedPositionAdded(_wrappedPosition);
+    }
+
+    /// @notice Allows the defaulter to mint wrapped tokens (Covered position) by
+    ///         sending the de-pegged token to the contract.
+    /// @dev    a) Only allow minting the covered position when the derived tranche got expired otherwise revert.
+    ///         b) Sufficient allowance of the principal token (i.e tranche) should be provided
+    ///            to the contract by the `msg.sender` to make execution successful.
+    /// @param  _amount Amount of covered position / wrapped token `msg.sender` wants to mint.
+    /// @param  _expiration Timestamp at which the derived tranche would get expired.
+    /// @param  _wrappedPosition Address of the Wrapped position which is used to derive the tranche.
+    function mint(
+        uint256 _amount,
+        uint256 _expiration,
+        address _wrappedPosition,
+        PermitData calldata _permitCallData
+    ) external override {
+        require(isAllowedWp(_wrappedPosition), "WFP:INVALID_WP");
+        address _tranche = address(
+            _deriveTranche(_wrappedPosition, _expiration)
+        );
+        _usePermitData(_tranche, _permitCallData);
+        // Only allow minting when the position get expired.
+        require(
+            ITranche(_tranche).unlockTimestamp() < block.timestamp,
```

Comment:
There's no need to make this exta external call because you already have the unlock timestamp provided in calldata and validated by the create2 hash.